subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/checkstyle/CheckstylePluginVersionIntegrationTest.groovy

```diff
@@ -320,6 +344,54 @@ class CheckstylePluginVersionIntegrationTest extends MultiVersionIntegrationSpec
         !file("build/tmp/checkstyleMain/main.xml").exists()
     }
 
+    @IgnoreIf({ !isSarifSupported(version.toString()) })
+    def "output SARIF report only when SARIF report is enabled"() {
+        given:
+        goodCode()
+        buildFile << '''
+            tasks.withType(Checkstyle) {
+                reports {
+                    xml.required = false
+                    html.required = false
+                    sarif.required = true
+                }
+            }
+        '''.stripIndent()
+
+        when:
+        succeeds 'checkstyleMain'
+
+        then:
+        file("build/reports/checkstyle/main.sarif").exists()
+        !file("build/reports/checkstyle/main.xml").exists()
+        !file("build/tmp/checkstyleMain/main.xml").exists()
+        !file("build/reports/checkstyle/main.html").exists()
+        !file("build/tmp/checkstyleMain/main.html").exists()
+    }
+
+    @IgnoreIf({ isSarifSupported(version.toString()) })
+    def "fails when SARIF enabled on unsupported checkstyle versions"() {
+        given:
+        goodCode()
+        buildFile << '''
+            tasks.withType(Checkstyle) {
+                reports {
+                    xml.required = false
+                    html.required = false
+                    sarif.required = true
+                }
+            }
+        '''.stripIndent()
+
+        when:
+        fails 'checkstyleMain'
+
+        then:
+        def versionNumber = VersionNumber.parse(version.toString())
```

Comment:
:x: parent class already has a `versionNumber` member, no need to parse again here

Reply:
updated